api/builder.go

```diff
@@ -2,30 +2,47 @@ package api
 
 import (
 	"context"
+	"math/big"
 	"mergemock/rpc"
 
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/ethereum/go-ethereum/core/types"
 	gethRpc "github.com/ethereum/go-ethereum/rpc"
 	"github.com/ethereum/go-ethereum/trie"
+	"github.com/holiman/uint256"
 	"github.com/sirupsen/logrus"
 )
 
+//go:generate go run github.com/fjl/gencodec -type ExecutionPayloadHeaderV1 -field-override executionPayloadHeaderMarshalling -out gen_edh.go
+
+// ExecutableDataV1 structure described at https://github.com/ethereum/execution-apis/src/engine/specification.md
 type ExecutionPayloadHeaderV1 struct {
-	ParentHash       common.Hash     `json:"parentHash"`
-	FeeRecipient     common.Address  `json:"feeRecipient"`
-	StateRoot        Bytes32         `json:"stateRoot"`
-	ReceiptsRoot     Bytes32         `json:"receiptsRoot"`
-	LogsBloom        Bytes256        `json:"logsBloom"`
-	PrevRandao       Bytes32         `json:"prevRandao"`
-	BlockNumber      Uint64Quantity  `json:"blockNumber"`
-	GasLimit         Uint64Quantity  `json:"gasLimit"`
-	GasUsed          Uint64Quantity  `json:"gasUsed"`
-	Timestamp        Uint64Quantity  `json:"timestamp"`
-	ExtraData        BytesMax32      `json:"extraData"`
-	BaseFeePerGas    Uint256Quantity `json:"baseFeePerGas"`
-	BlockHash        common.Hash     `json:"blockHash"`
-	TransactionsRoot Bytes32         `json:"transactionsRoot"`
+	ParentHash       common.Hash    `json:"parentHash"    gencodec:"required"`
+	FeeRecipient     common.Address `json:"feeRecipient"  gencodec:"required"`
+	StateRoot        common.Hash    `json:"stateRoot"     gencodec:"required"`
+	ReceiptsRoot     common.Hash    `json:"receiptsRoot"  gencodec:"required"`
+	LogsBloom        []byte         `json:"logsBloom"     gencodec:"required"`
+	Random           common.Hash    `json:"prevRandao"    gencodec:"required"`
+	Number           uint64         `json:"blockNumber"   gencodec:"required"`
+	GasLimit         uint64         `json:"gasLimit"      gencodec:"required"`
+	GasUsed          uint64         `json:"gasUsed"       gencodec:"required"`
+	Timestamp        uint64         `json:"timestamp"     gencodec:"required"`
+	ExtraData        []byte         `json:"extraData"     gencodec:"required"`
+	BaseFeePerGas    *big.Int       `json:"baseFeePerGas" gencodec:"required"`
+	BlockHash        common.Hash    `json:"blockHash"     gencodec:"required"`
+	TransactionsRoot common.Hash    `json:"transactionsRoot"  gencodec:"required"`
+}
+
+// JSON type overrides for executableData.
+type executionPayloadHeaderMarshalling struct {
```

Comment:
I think these type overrides are a horrible hack, but it's also nice to have `uint64` in the interface ofc., so it's ok-ish

Reply:
yeah :(